service/src/test/java/tds/assessment/web/endpoints/AdminSubjectControllerIntegrationTests.java

```diff
@@ -1,45 +1,54 @@
 package tds.assessment.web.endpoints;
 
-import com.jayway.restassured.http.ContentType;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.springframework.boot.test.IntegrationTest;
-import org.springframework.boot.test.SpringApplicationConfiguration;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
-import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
 
-import tds.assessment.AssessmentServiceApplication;
+import java.net.URI;
+import java.util.Optional;
 
-import static com.jayway.restassured.RestAssured.given;
-import static org.hamcrest.Matchers.equalTo;
+import tds.assessment.SetOfAdminSubject;
+import tds.assessment.services.AdminSubjectService;
 
-@RunWith(SpringJUnit4ClassRunner.class)
-@SpringApplicationConfiguration(classes = AssessmentServiceApplication.class)
-@WebAppConfiguration
-@IntegrationTest("server.port:8080")
+import static org.hamcrest.CoreMatchers.is;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(AdminSubjectController.class)
```

Comment:
Looking through the Spring documentation I think this is the best way we can integration test controller logic.  The @WebMvcTest annotation makes Spring just stand up the Controller layer.  This would include the message converters, advice, and rest controllers.  Note that I use @MockBean and mockito to mock out the service.  

Main benefit here is that it is not hitting the database nor standing up the full Web container (so no port 8080 conflict).